### ISSUE TITLE
Rotation and flipping

### ISSUE THREAD

**Issue by jbox144** [CircuitVerse/CircuitVerse#108]
It's kind of annoying how porly supported rotation is. I'd also like to see and option to support fliping some components too. Here are a few major suspects:

- [ ] **Power and ground**
Aside from the power and ground pins being absolute hell to actually manipulate thanks to their broken selection area, you can't even rotate them! I demand a refund!

- [ ] **D-Flip, D-Latch, T-Flip, JK-Flip, SR-Flip**
I often find myself in situation where I have to do, in technical terms, "spaghetti wire" to achieve a circuit design. This makes the circuitry rather illegible in circumstance.

- [ ] **Subcircuits**

- [ ] **Various others**
I'll see if I can find any other parts which can't be rotated but should certainly be able to

Also, as stated, I'd like to be able to flip parts around. An example would be something like the Multiplexer. Often times the selection pin is in a weird location, which again leads to "spaghetti logic". The same can be said for the splitter, although luckily to a less annoying extent, but still annoying all the same.

**Comment by satu0king** [comment]
I agree this is an issue, the developers decided to fix the rotation of certain elements to keep things simple. 

Fixing the rotation of Ground and Power is for purely aesthetic reasons but we can perhaps relax it now. 

I agree flipping is very important and has to be looked into. Especially with the plexers and splitters.

**Comment by sanctophagus** [comment]
Hey @jbox144 !
I'd like to work on this issue, if anyone else ain't working on it.

**Comment by jbox144** [comment]
@sancophagus

sure!

**Comment by Parikshit-Hooda** [comment]
@sanctophagus What's the progress on this?

**Comment by shubhwal56** [comment]
@tachyons Can I work on this issue?
